std/conv.d

```diff
@@ -3901,136 +3901,81 @@ Furthermore, emplaceRef optionally takes a type paremeter, which specifies
 the type we want to build. This helps to build qualified objects on mutable
 buffer, without breaking the type system with unsafe casts.
 +/
-package void emplaceRef(UT, Args...)(ref UT chunk, auto ref Args args)
-if (is(UT == Unqual!UT))
-{
-    emplaceRef!(UT, UT)(chunk, args);
-}
-// ditto
 package void emplaceRef(T, UT, Args...)(ref UT chunk, auto ref Args args)
 if (is(UT == Unqual!T))
 {
-    emplaceImpl!T(chunk, args);
-}
-
-private template emplaceImpl(T)
-{
-    alias UT = Unqual!T;
-
-    void emplaceImpl()(ref UT chunk)
+    static if (args.length == 0)
     {
         static assert (is(typeof({static T i;})),
             convFormat("Cannot emplace a %1$s because %1$s.this() is annotated with @disable.", T.stringof));
         emplaceInitializer(chunk);
     }
-
-    // Primitive types, enums, static arrays, dynamic arrays
-    static if (!is(T == struct))
-    void emplaceImpl(Arg)(ref UT chunk, auto ref Arg arg)
+    else static if (
+        !is(T == struct) && Args.length == 1 /* primitives, enums, arrays */
```

Comment:
Yah, it occurred to me too. That would be an enhancement. I've added a test case for unions (see my next commit) and they seem to work. Could you please post a few tests that fail?